mcp/src/minecraft/net/minecraft/block/Block.java

```diff
@@ -1360,22 +1361,22 @@ public static void registerBlocks()
-        registerBlock(501, "amber_ore", (new Block(Material.ROCK)).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreAmber").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
-        registerBlock(502, "tanzanite_ore", (new Block(Material.ROCK)).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreTanzanite").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
+        registerBlock(500, "jade_ore", (new BlockCustomOre()).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreJade").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
+        registerBlock(501, "amber_ore", (new BlockCustomOre()).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreAmber").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
+        registerBlock(502, "tanzanite_ore", (new BlockCustomOre()).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreTanzanite").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
         registerBlock(503, "permanent_dirt", (new Block(Material.ROCK)).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("permanentDirt")); // Deprecated in favor of coarse dirt
         registerBlock(504, "luindol", (new BlockCustomFlower(BlockCustomFlower.generateCrossBoundingBox(12, 19))).setUnlocalizedName("luindol"));
         registerBlock(505, "hopper_mushroom", (new BlockCustomFlower(BlockCustomFlower.generateCrossBoundingBox(4, 11, 16))).setUnlocalizedName("hopperMushroom"));
-        registerBlock(506, "amethyst_ore", (new Block(Material.ROCK)).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreAmethyst").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
-        registerBlock(507, "ruby_ore", (new Block(Material.ROCK)).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreRuby").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
-        registerBlock(508, "onyx_ore", (new Block(Material.ROCK)).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreOnyx").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
-        registerBlock(509, "moonstone_ore", (new Block(Material.ROCK)).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreMoonstone").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
+        registerBlock(506, "amethyst_ore", (new BlockCustomOre()).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreAmethyst").setCreativeTab(CreativeTabs.BUILDING_BLOCKS));
+        registerBlock(507, "ruby_ore", (new Block(Material.ROCK)).setHardness(1.5F).setResistance(10.0F).setSoundType(SoundType.STONE).setUnlocalizedName("oreRuby").setCreativeTab(CreativeTabs.BUILDING_BLOCKS)); //needs ruby item for furnace recipes and item drop
```

Comment:
It looked like minecraft was going to add rubies for a while there (they ended up becoming emeralds). This is why we don't have a ruby item at the moment, it can definitely be added to our todo list.